lib/stat_calculator.rb

```diff
@@ -171,4 +171,68 @@ def worst_offense
     worst_offense = @teams.find { |team| team.team_id.to_s == worst_offense_team.to_s}
     worst_offense.team_name
   end
+
+  def most_tackles(season)
+    game_ids_for_season = []
+    @games.each do |game|
+      if game.season == season
+        game_ids_for_season << game.game_id
+      end
+    end
+   
+    team_tackles = {}
+    @game_teams.each do |game_team|
+      if game_ids_for_season.include?(game_team.game_id)
+        team_tackles[game_team.team_id] ||= 0
+        team_tackles[game_team.team_id] += game_team.tackles
+      end
+    end
+    
+    team_with_most_tackles = nil
+    most_tackles = 0
+    team_tackles.each do |team_id, tackles|
+      if tackles > most_tackles
+        most_tackles = tackles
+        team_with_most_tackles = team_id
+      end
+    end
+    
+    @teams.each do |team|
+      if team.team_id.to_s == team_with_most_tackles.to_s
+        return team.team_name
+      end
+    end
+  end  
+
+  def fewest_tackles(season)
+    game_ids_for_season = []
+    @games.each do |game|
+      if game.season == season
+        game_ids_for_season << game.game_id
+      end
+    end
+   
+    team_tackles = {}
+    @game_teams.each do |game_team|
+      if game_ids_for_season.include?(game_team.game_id)
+        team_tackles[game_team.team_id] ||= 0
+        team_tackles[game_team.team_id] += game_team.tackles
+      end
+    end
+    
+    team_with_least_tackles = nil
+    least_tackles = 1000000 #random number I chose, just needed a big enough number
```

Comment:
Same here:
- may be able to accomplish this method with a min_by in this block

Out of curiosity, what is the purpose of setting least_tackles to 100000?
Instead, could setting it to the first teams number of tackles work?

Reply:
The purpose of setting it to a high number was to ensure that teams tackle count will be smaller. 
I do realize there is room for refactoring. I was under the impression we were trying to get working code first. This is the just way my brain was able to put all the pieces of the puzzle together.

Reply:
Nope, it's all good. Like I said in my main review comment. I'm ok with merging this. Just wanted to spark some refactor ideas for you for later!

Reply:
I appreciate the advice and the review!